### ISSUE TITLE
Edge artifact with reflect mode

### ISSUE THREAD

**Issue by not522** [scikit-image/scikit-image#2990]
Default edge mode changed from `constant` to `reflect` for `resize` and `rescale` by #2484.
But, I find the following artifact around the edge of the image with `reflect` mode.
I recommend `edge` to default mode.

```
import matplotlib.pyplot as plt
import numpy as np
import skimage.transform

a = np.array([[1, 1, 1], [1, 0, 1], [1, 1, 1]], dtype=np.float64)
b = skimage.transform.rescale(a, (5, 5), mode='reflect')
plt.axes().set_aspect('equal')
plt.pcolor(b, cmap=plt.cm.binary)
plt.show()
```
![reflect](https://user-images.githubusercontent.com/1255726/37760381-d6699942-2df9-11e8-876d-e17af6d6590e.png)
`edge` mode does not make the artifact.
![edge](https://user-images.githubusercontent.com/1255726/37760389-db3bb248-2df9-11e8-8581-50cbd6a43033.png)

- Operating system and version
macOS High Sierra 10.13.3
- Python version
3.6.2
- scikit-image version
0.13.1

**Comment by grlee77** [comment]
In #1514 we decided `reflect` was a preferable default to `constant` as the majority of natural images are not zero at their boundaries.

I think `edge` is also preferable to `constant` as a default, but not necessary to `reflect`.  I think which is more desirable is going to depend on the particular choice of image.  What you have referred to as "artifact" here is the expected behavior for that edge mode.  None of these choices is going to be optimal in all cases (thus the presence of the option in the first place).

I don't have a strong preference for `reflect` over `edge`, but given that we already implemented and started the deprecation cycle based on `reflect`, I am not inclined to change it again to `edge` at this point unless there is a stronger reason to do so.

**Comment by jni** [comment]
> given that we already implemented and started the deprecation cycle based on reflect, I am not inclined to change it again to edge at this point unless there is a stronger reason to do so.

I agree with this, thanks @grlee77. If we have another core dev on this camp, I'll go ahead and close this.

@not522 Thank you very much for the report. As @grlee77 said, the optimal choice is really use-dependent. For your use, you should use `mode='edge'`. For other uses (for example to expand borders or augment natural image data for neural networks), reflect is a better choice. Therefore I think we will probably leave the default as is.

**Comment by not522** [comment]
I believe `edge` mode is **always** better than `reflect`.
I will add code for explanation later, please wait.

**Comment by not522** [comment]
The following code shows how to specify the index by the edge modes.
`reflect` mode uses the inner index to interpolate the outside of the image.
I think there is no reason to use the inner index for the outside.
`edge` mode always uses closer index than `reflect` mode.
```
import matplotlib.pyplot as plt
import numpy as np
import skimage.transform

a = np.arange(10, dtype=np.float64).reshape(1, 10)
modes = ['constant', 'edge', 'reflect']
outputs = {}
for mode in modes:
    outputs[mode] = skimage.transform.rescale(a, (1, 1000), mode=mode).squeeze()
for mode, output in outputs.items():
    plt.plot(output, label=mode)
plt.legend(loc=2)
plt.savefig('index.png')
```
![index](https://user-images.githubusercontent.com/1255726/37806347-ca6277de-2e82-11e8-8978-09711dce4d7f.png)
In #1514, `reflect` was recommended because it is similar to the behavior of OpenCV' `resize`, but it is same as `edge` actually.
This is another reason I recommend `edge` mode.
```
import cv2
import matplotlib.pyplot as plt
import numpy as np

a = np.arange(10, dtype=np.float64).reshape(1, 10)
plt.plot(cv2.resize(a, (10000, 1)).squeeze(), label='OpenCV')
plt.legend(loc=2)
plt.savefig('opencv.png')
```
![opencv](https://user-images.githubusercontent.com/1255726/37806340-c23f5888-2e82-11e8-814e-ee911b2ae227.png)
By the way, if maintainers think this problem is acceptable, I agree with the opinion.
